src/main/scala/lms/core/codegen/C_codegen.scala

```diff
@@ -396,6 +396,8 @@ class ExtendedCCodeGen extends CompactScalaCodeGen with ExtendedCodeGen {
     case n @ Node(s,"array_resize", List(arr, nsize), _) =>
       val tpe = remap(typeMap.get(s).map(_.typeArguments.head).getOrElse(manifest[Unknown]))
       emit(s"($tpe*)realloc("); shallow(arr); emit(", "); shallow1(nsize, precedence("*")); emit(s" * sizeof($tpe))")
+    case n @ Node(s, "array-of-char-to-string", List(x:Sym), _) =>
+      shallow(x)
```

Comment:
this is currently wrong. Should have an array-copy first, the return the copied array. However, array_length node code-gen is still missing, making array-copy unfeasible. Will fix later.

Reply:
not sure I follow what the flare code is doing. :) Is it building a string?

I was trying to copy the array but hit a ??? when fetch array-length. I guess that needs to be fixed anyway.

Reply:
No... this is not defined in C, so there is no implementation possible. I am just showing you functions that have the length given as argument.